src/modules/navigation/hooks/useSharedDriveLink.tsx

```diff
@@ -26,7 +26,14 @@ const useSharedDriveLink = (sharing: SharedDrive): UseFileLinkResult => {
   const cozyUrl = client?.getStackClient().uri as string
 
   const app = 'drive'
-  const path = `shareddrive/${sharing._id}/${sharing.rules[0].values[0]}`
+
+  /** Set shared drive path
+   * if user is owner of this shared drive, the path should be `folder/:folderId`
+   * otherwise the path should be `shareddrive/:driveId/:folderId`
+   */
+  const path = sharing.owner
+    ? `folder/${sharing.rules[0].values[0]}`
+    : `shareddrive/${sharing._id}/${sharing.rules[0].values[0]}`
```

Comment:
Agreeing with the general philosophy, but I know Christophe also worked on it friday and ended up with a slightly different solution, let's wait him to compare both solutions